### PR TITLE
CONSOLE-4104: Implement basic OLM V1 catalog view with filtering

### PR DESCRIPTION
**Pull request by TheRealJon**
screencast-localhost_9000-2024 08 07-12_38_19

### CODE REVIEW

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @reestr

**Comment by TheRealJon** [comment]
/hold 

Working on a bug

File: pkg/server/server.go
Comment (yapei) [inline pkg/server/server.go:785]: It looks like we have issue here
```
[yapei@yapei-mac TheRealJon-console (CONSOLE-4101)]$ ./bin/bridge       
W0717 16:02:19.596299   58764 authoptions.go:112] Flag inactivity-timeout is set to less then 300 seconds and will be ignored!
W0717 16:02:19.596368   58764 authoptions.go:263] running with AUTHENTICATION DISABLED -- for development use only!
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10fbfdaed]

goroutine 1 [running]:
github.com/openshift/console/pkg/proxy.NewProxy(...)
	/Users/yapei/go/src/github.com/openshift/TheRealJon-console/pkg/proxy/proxy.go:65
github.com/openshift/console/pkg/server.(*Server).CatalogdHandler(0xc0001a7888)
	/Users/yapei/go/src/github.com/openshift/TheRealJon-console/pkg/server/server.go:786 +0xcd
github.com/openshift/console/pkg/server.(*Server).HTTPHandler(0xc0001a7888)
	/Users/yapei/go/src/github.com/openshift/TheRealJon-console/pkg/server/server.go:312 +0xe45
main.main()
	/Users/yapei/go/src/github.com/openshift/TheRealJon-console/cmd/bridge/main.go:590 +0x4fa6

```

**Comment by TheRealJon** [comment]
/hold cancel

@yapei I've addressed the issue in bridge.

There may be a couple of other bugs, but since we are approaching the 4.18 branch date we decided we'd like to try and get this merged and address those via backport.

**Comment by yapei** [comment]
some observations during testing:
1. user is unable to install operator because console will return `not found` page after clicking on `Install` button(even when `TechPreviewNoUpgrade` is not enabled)

https://github.com/user-attachments/assets/3c953cad-6f7a-4b91-a52f-f2e9f7b726ce


2. translation issues
Screenshot 2024-07-19 at 12 25 55 PM

3. `Extension Catalog` page doesn't show catalog items 
```
$ oc get featuregate cluster -o json | jq .spec.featureSet
"TechPreviewNoUpgrade"
$ oc get clustercatalog
NAME            PHASE      AGE
catalog-74924   Unpacked   12m
```
Screenshot 2024-07-19 at 12 20 13 PM

**Comment by reestr** [comment]
lgtm

File: frontend/packages/operator-lifecycle-manager-v1/locales/en/olm-v1.json
Comment (opayne1) [inline frontend/packages/operator-lifecycle-manager-v1/locales/en/olm-v1.json:7]: I know the console uses "please" in some messages, but thought it might be worth noting that we avoid it in the documentation.
```suggestion
  "Check that OLM v1 is configured and that you have created at least one valid ClusterCatalog.": "Check that OLM v1 is configured and that you have created at least one valid ClusterCatalog."
```

**Comment by opayne1** [comment]
One small docs comment but otherwise lgtm

/label docs-approved

File: frontend/packages/operator-lifecycle-manager-v1/locales/en/olm-v1.json
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager-v1/locales/en/olm-v1.json]: Thanks, I'll update that.

File: frontend/packages/operator-lifecycle-manager-v1/src/database/packages.ts
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager-v1/src/database/packages.ts:23]:
```suggestion
    channels,
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts:61]:
```suggestion
  Proxy: InfraFeatures.proxyAware,
```

**Review by jhadvig — APPROVED** [review]
Nice job @TheRealJon 👍 Thanks for all the hard work on this.
/lgtm

**Comment by jhadvig** [comment]
Overiding the test due to CI infrastructure issues. The test already [passed](https://github.com/openshift/console/pull/14056#issuecomment-2274345875) with current changes.

/override ci/prow/analyze ci/prow/backend ci/prow/e2e-gcp-console ci/prow/frontend ci/prow/images

**Pull request merged**
